platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js

```diff
@@ -55,17 +56,28 @@ class ViewerRetrieveStudyData extends Component {
           );
         }
         studyMetadata.setDisplaySets(study.displaySets);
+
         // Updates WADO-RS metaDataManager
         updateMetaDataManager(study);
         studyMetadataManager.add(studyMetadata);
+
         // Attempt to load remaning series if any
-        this._attemptToLoadRemainingSeries(studyMetadata);
+        this._loadRemainingSeries(studyMetadata).then(
+          this._studyDidLoad.bind(this)
```

Comment:
Its a little cleaner to bind this in the constructor:

`this._studyDidLoad = this._studyDidLoad.bind(this);`

Then you can just call `this._studyDidLoad()` in the class methods.

Reply:
Good tip. Just updated the code.